### PR TITLE
SALTO-6954: Removing some async calls from validators

### PR DESCRIPTION
**Pull request by yelly**
Won't solve all the problems but hopefully will lead to some performance gains.

---

_Additional context for reviewer_

---

_Release Notes_: 
None.

---

_User Notifications_: 
None.

### CODE REVIEW

File: packages/salesforce-adapter/src/change_validators/map_keys.ts [hunk 1]
```diff
@@ -44,72 +42,72 @@ const isNum = (str: string | undefined): boolean => !_.isEmpty(str) && !Number.i
...
-          const previewPrefix = actualPath.slice(0, actualPath.findIndex((val, idx) => val !== expectedPath[idx]) + 1)
-          if (!_.isEqual(actualPath, expectedPath)) {
-            errors.push({
-              elemID: after.elemID.createNestedID(fieldName, ...previewPrefix),
-              severity: 'Error',
-              message: 'Incorrect map key in field',
-              detailedMessage: `${typeName} ${after.value.fullName} field ${fieldName}: Incorrect map key ${actualPath?.join(API_NAME_SEPARATOR)}, should be ${expectedPath.join(API_NAME_SEPARATOR)}`,
-            })
-          }
+  await awu(
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/change_validators/map_keys.ts:48]: We can avoid the async code here by using `transformValuesSync`

**Review by tamtamirr — COMMENTED** [review]
LGTM! We can still avoid more promises. See my comment

File: packages/salesforce-adapter/src/change_validators/map_keys.ts (hunk 1 above)
Comment (yelly) [reply packages/salesforce-adapter/src/change_validators/map_keys.ts]: Done.

**Review by tamtamirr — APPROVED** [review]
LGTM

**Pull request merged**